HPCC-18086 Improve query dll copying/naming scheme

afishbeck opened the pull request:
Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

afishbeck committed:
HPCC-18086 Improve query dll copying/naming scheme

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 0d934560062538a0c2734934a8b6f15ba9b71ff3
Build: success
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 93.385 seconds |


Install hpccsystems-platform-community_6.4.1-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |95 |95 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |733 |733 |0 |
|test (thor) |627 |627 |0 |
|test (roxie) |760 |760 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

richardkchapman commented:
@ghalliday Please review

ghalliday commented on esp/services/ws_workunits/ws_workunitsService.cpp:
Unlikely:  This write should probably be outside the try block - otherwise on disk full the first write will fail, on the second time it will rename the empty file.

ghalliday reviewed (commented):
Generally looks solid.  @afishbeck one minor issue.

afishbeck committed:
HPCC-18086 Copy workunit dll shouldn't continue write temp file fails

Don't catch the exception and ontinue if writing of the temporary
dll fails.  This would most likely be from some fatal condition
such as disk is full.

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

afishbeck commented:
@ghalliday Updated based on your comment.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: fbfce6a73afb8b156981bec3bcdaf1e12a7a0636
Build: success
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 93.194 seconds |


Install hpccsystems-platform-community_6.4.2-rc1.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |96 |96 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |733 |733 |0 |
|test (thor) |627 |627 |0 |
|test (roxie) |760 |760 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

ghalliday commented:
@afishbeck probably best to squash
@richardkchapman looks good to merge.  I'll leave to you since 6.4.2.
